src/NHibernate.Test/Async/Criteria/Lambda/FunctionsIntegrationFixture.cs

```diff
@@ -22,7 +22,7 @@ public class FunctionsIntegrationFixtureAsync : TestCase
 	{
 		protected override bool AppliesTo(Dialect.Dialect dialect)
 		{
-			return !(dialect is AbstractHanaDialect); // HANA does not support inserting a row without specifying any column values
+			return TestDialect.SupportsEmptyInserts;
```

Comment:
This surprises me, because we already test a database which is not supposed to support this, and which does not fail in all these tests modified by this PR. I have to check what is going on.

Reply:
Ok, the test uses the `native` generator, which resolves to `identity` if it is supported, otherwise `sequence` if supported, otherwise `hilo`. The other db not supporting empty inserts does override `NativeIdentifierGeneratorClass` for using `hilo`, thus it has a value to insert.

Then the exact condition here should be `TestDialect.SupportsEmptyInserts || dialect.NativeIdentifierGeneratorClass != typeof(IdentityGenerator)`.

Reply:
Shouldn't the check for the `NativeIdentifierGeneratorClass` be included in `TestDialect.SupportsEmptyInserts` then? It seems strange that you'd have to check for `TestDialect.SupportsEmptyInserts` and basically everywhere add the check for the `NativeIdentifierGeneratorClass` right after it. If `TestDialect.SupportsEmptyInserts` really means something else, would it make sense to introduce a new property like `TestDialect.SupportsEmptyInsertsWithNativeIdentifierGenerator`?

Reply:
If the mapped class was using `identity` directly instead, testing on `NativeIdentifierGeneratorClass` inside `SupportsEmptyInserts ` would defeat its purpose.

The test here does empty inserts only if the `native` generator does resolves to `identity`, otherwise it does not perform empty inserts. So when the dialect does not support empty inserts but does not resolves `native` to `identity`, it is irrelevant to disable that test.

`SupportsEmptyInsertsWithNativeIdentifierGenerator` is not meaningful. But since the case is very frequent, why not adding a `SupportsEmptyInsertsOrHasNonIdentityNativeGenerator`. It could be something like:

```c#
/// <summary>
/// Either supports inserting in a table without any column specified in the insert, or has a native
/// generator strategy resolving to something else than identity.
/// </summary>
/// <remarks>This property is useful for cases where empty inserts happens only when the entities
/// generator is <c>native</c> while the dialect uses <c>identity</c> for this generator.</remarks>
public bool SupportsEmptyInsertsOrHasNonIdentityNativeGenerator
	=> SupportsEmptyInserts || _dialect.NativeIdentifierGeneratorClass != typeof(IdentityGenerator);
```